plugins/node/opentelemetry-instrumentation-express/src/instrumentation.ts

```diff
@@ -202,9 +211,14 @@ export class ExpressInstrumentation extends InstrumentationBase<
             ExpressLayerType.REQUEST_HANDLER &&
           rpcMetadata?.type === RPCType.HTTP
         ) {
-          rpcMetadata.span.updateName(
+          const name = instrumentation._getSpanName(
+            {
+              request: req,
+              route,
+            },
             `${req.method} ${route.length > 0 ? route : '/'}`
           );
+          rpcMetadata.span.updateName(name);
```

Comment:
Here we actually updating the name of the `http` span to [match the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#name):
> Many REST APIs encode parameters into URI path, e.g. /api/users/123 where 123 is a user id, which creates high cardinality value space not suitable for span names. In case of HTTP servers, these endpoints are often mapped by the server frameworks to more concise HTTP routes, e.g. /api/users/{user_id}, which are recommended as the low cardinality span names

Do we intend this new hook to affect the `http` span name as well? If we do then I suggest to make it more clear in the docs and the hook parameters that this is the case.

Reply:
Yes it is intentional (and perhaps even the most important part of the hook). Added a comment to `types.ts`

Reply:
Thanks :)